tsfresh/feature_extraction/extraction.py

```diff
@@ -190,32 +195,69 @@ def _extract_features_parallel_per_sample(kind_to_df_map, settings, column_id, c
                                                            column_value=column_value,
                                                            settings=settings)
     pool = Pool(settings.n_processes)
+    total_number_of_expected_results = 0
 
     # Submit map jobs per kind per sample
     results_fifo = Queue()
+
     for kind, df_kind in kind_to_df_map.items():
         df_grouped_by_id = df_kind.groupby(column_id)
+
+        total_number_of_expected_results += len(df_grouped_by_id)
+
+        chunksize = _calculate_best_chunksize(df_grouped_by_id, settings)
+
         results_fifo.put(
-            pool.map_async(
+            pool.imap_unordered(
                 partial_extract_features_for_one_time_series,
                 [(kind, df_group) for _, df_group in df_grouped_by_id],
-                chunksize=settings.chunksize
+                chunksize=chunksize
             )
         )
 
     pool.close()
 
     # Wait for the jobs to complete and concatenate the partial results
     dfs_per_kind = []
-    while not results_fifo.empty():
-        map_result = results_fifo.get()
-        dfs = map_result.get()
-        dfs_per_kind.append(pd.concat(dfs, axis=0).astype(np.float64))
 
-    result = pd.concat(dfs_per_kind, axis=1).astype(np.float64)
+    # Do this all with a progress bar
+    with tqdm(total=total_number_of_expected_results, desc="Feature Extraction", disable=settings.disable_progressbar) as progress_bar:
+        # We need some sort of measure, when a new result is there. So we wrap the
+        # map_results into another iterable which updates the progress bar each time,
+        # a new result is there
+        def iterable_with_tqdm_update(queue, progress_bar):
+            for element in queue:
+                progress_bar.update(1)
+                yield element
 
-    pool.join()
-    return result
+        while not results_fifo.empty():
+            map_result = results_fifo.get()
+            dfs = iterable_with_tqdm_update(map_result, progress_bar)
+            dfs_per_kind.append(pd.concat(dfs, axis=0).astype(np.float64))
+
+        result = pd.concat(dfs_per_kind, axis=1).astype(np.float64)
+
+        pool.join()
+        return result
+
+
+def _calculate_best_chunksize(iterable_list, settings):
+    """
+    Helper function to calculate the best chunksize for a given number of elements to calculate,
+    or use the one in the settings object.
+
+    The formula is more or less an empirical result.
+    :param iterable_list: A list which defines how many calculations there need to be.
+    :param settings: The settings object where the chunksize may already be given (or not).
+    :return: The chunksize which should be used.
+    """
+    if not settings.chunksize:
+        chunksize, extra = divmod(len(iterable_list), settings.n_processes * 5)
```

Comment:
maybe add a todo here that we investigate the optimal chunksize this in the future